source/simulator/core.cc

```diff
@@ -568,21 +568,6 @@ namespace aspect
     if (particle_world.get() != nullptr)
       particle_world->backup_particles();
 
-    // Re-compute the pressure scaling factor. In some sense, it would be nice
-    // if we did this not just once per time step, but once for each solve --
-    // i.e., multiple times per time step if we iterate out the nonlinearity
-    // during a Newton or Picard iteration. But that's more work,
-    // and the function would need to be called in more different places.
-    // Unless we have evidence that that's necessary, let's assume that
-    // the reference viscosity does not change too much between nonlinear
-    // iterations and that it's ok to update it only once per time step.
-    //
-    // The call to this function must precede the one to the computation
-    // of constraints because some kinds of constraints require scaling
-    // pressure degrees of freedom to a size adjusted by the pressure_scaling
-    // factor.
-    pressure_scaling = compute_pressure_scaling_factor();
-
     // then interpolate the current boundary velocities. copy constraints
```

Comment:
I don't understand how this works, you need to update the factor before setting the constraints right below or not? Based on this comment it seems like you always have to redo the constraints after changing the pressure scaling, which you do not. Or am I missing something?